### ISSUE TITLE
Examples, singular or plural?

### ISSUE THREAD

**Issue by nleiva** [golang-standards/project-layout#7]
The [examples](https://github.com/golang-standards/project-layout/tree/master/examples) folder seems to conflict with: [Use singular form for collection repo/folder name](https://dmitri.shuralyov.com/idiomatic-go#use-singular-form-for-collection-repo-folder-name). Thoughts?

**Comment by kcq** [comment]
You are absolutely right! However, even the Go creators deviate from these guidelines. The 'errors', 'strings' and 'bytes' packages are some of those examples.

Either way, it'll be good to add a few references around naming:
* https://github.com/golang/go/wiki/CodeReviewComments
* https://blog.golang.org/package-names
* https://golang.org/doc/effective_go.html#names

**Comment by nleiva** [comment]
Look at [comment-2](https://dmitri.shuralyov.com/idiomatic-go/entries/5#comment-2). `example(s)` could be a collection of them (as in [packer](https://github.com/hashicorp/packer/tree/master/examples)), unlike `errors`, `strings`, and `bytes`. The `encoding` example is an excellent reference (`encoding/json`, `encoding/xml`, `encoding/gob`).

I agree people deviate from this one. Just wondering what we would consider as best practice.

**Comment by kcq** [comment]
If it's not a package name/directory it's ok to deviate from that recommendation/guideline. The `examples` directory falls into this category :-) Ultimately it's up to you to decide what makes sense for your project.

**Comment by mihai-chiorean** [comment]
in the exceptions to the norm of using singular, provided above, wouldn't "string" (as opposed to "strings") conflict with the reserved keyword `string`? same for errors and bytes. I think that may be the rationale behind the exception to use plural
